src/main/java/org/logstash/input/DeadLetterQueueInputPlugin.java

```diff
@@ -44,12 +44,15 @@
     private final AtomicBoolean open;
     private final Timestamp targetTimestamp;
 
+    private boolean readerHasState;
+
     public DeadLetterQueueInputPlugin(Path path, boolean commitOffsets, Path sinceDbPath, Timestamp targetTimestamp) throws Exception {
         this.queueReader = new DeadLetterQueueReader(path);
         this.commitOffsets = commitOffsets;
         this.open = new AtomicBoolean(true);
         this.sinceDbPath = sinceDbPath;
         this.targetTimestamp = targetTimestamp;
+        this.readerHasState = false;
```

Comment:
The plugin instance is used by multiple threads, so also if boolean is a 1 word operation, and the update could be seen as an atomic write, we however need to pass the memory barrier to publish the value to other threads.
- or we add the `volatile` modifier
- or as similarly done by `open` flag, it has to be an AtomicBoolean

Reply:
I've added volatile in the new push. This should be sufficient because we don't set the variable based on the variable itself, so there are no race conditions that would require a lock or use of atomic.

Reply:
The `open` flag has  a similar usage pattern, so maybe is best to keep the same solution and switch from `volatile` to `AtomicBoolean` also for the `readerHasState`

Reply:
In the most recent push I've changed it to be an `AtomicBoolean` similar to `open` (declared in same line, let me know if you don't like that).